src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -628,9 +638,19 @@ public TResult[] ToArray()
                 }
                 return results;
             }
+
+            public List<TResult> ToList()
+            {
+                var results = new List<TResult>(_source.Count);
+                for (int i = 0; i < _source.Count; i++)
+                {
```

Comment:
(You probably want to move the setting of cnt to before constructing the list so you can use the cnt in the ctor call as well.)